sources/packages/api/src/services/institution/institution.service.ts

```diff
@@ -396,4 +396,44 @@ export class InstitutionService extends RecordDataModelService<Institution> {
       .where("user.userName = :userName", { userName })
       .getOne();
   }
+
+  async deleteAssociationByUserID(
+    institutionUser: InstitutionUser
+  ): Promise<void> {
+    const previousAssociations = await this.institutionUserAuthRepo.find({institutionUser:institutionUser});
+    await this.institutionUserAuthRepo.remove(previousAssociations);
+  }
+
+  async updateInstitutionUser(
+    permissionInfo: InstitutionUserPermissionDto,
+    institutionUser: InstitutionUser
+  ): Promise<InstitutionUserAuth[]> {
+   let newAuthorizationEntries = [] as InstitutionUserAuth[]
+    // Create the permissions for the user under the institution.
+    for (const permission of permissionInfo.permissions) {
+      const newAuthorization = new InstitutionUserAuth();
+      newAuthorization.institutionUser = institutionUser
+      if (permission.locationId) {
+        // Add a location specific permission.
+        newAuthorization.location = {
+          id: permission.locationId,
+        } as InstitutionLocation;
+      }
+      // Find the correct user type and role.
+      const authType = await this.institutionUserTypeAndRoleRepo.findOne({
+        type: permission.userType,
+        role: permission.userRole ?? null,
+      });
+      if (!authType) {
+        throw new Error(
+          "The combination of user type and user role is not valid.",
+        );
+      }
+      newAuthorization.authType = authType;
+      newAuthorizationEntries.push(newAuthorization)
+    }
+    // delete existing associations
+    await this.deleteAssociationByUserID(institutionUser)
```

Comment:
I believe that it is possible to execute both commands together.
It would be good to have them executed in "same transaction" in case of the save fail (line 437), all the authentication records would be lost.